nextcord/automod.py

```diff
@@ -0,0 +1,351 @@
+
+    async def delete(self):
+        """|coro|
+
+        Delete this auto moderation rule."""
+        await self._state.http.delete_automod_rule(guild_id=self.guild.id, rule_id=self.id)
+
+    @property
+    def creator(self) -> Optional[Member]:
+        """Optional[:class:`Member`]: The member that created this rule."""
+        return self.guild.get_member(self.creator_id)
+
+    @property
+    def exempt_roles(self) -> List[Role]:
+        """List[:class:`Role`]: A list of role that will not be affected by this rule. `[]` if not set."""
+        return [self.guild.get_role(exempt_role_id) for exempt_role_id in self.exempt_role_ids]  # type: ignore -- they can't be None.
+
+    @property
+    def exempt_channels(self) -> List[GuildChannel]:
+        """List[:class:`GuildChannel`]: A list of channels that will not be affected by this rule. `[]` if not set."""
+        return [self.guild.get_channel(exempt_channel_id) for exempt_channel_id in self.exempt_channel_ids]  # type: ignore -- same
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = ...,
+        event_type: Optional[EventType] = ...,
+        keyword_filters: Optional[List[str]] = ...,
+        notify_channel: Optional[GuildChannel] = ...,
+        timeout_seconds: Optional[int] = ...,
+        enabled: Optional[bool] = ...,
+        exempt_roles: Optional[List[Role]] = ...,
+        exempt_channels: Optional[List[GuildChannel]] = ...,
+        reason: Optional[str] = ...,
+    ) -> AutoModerationRule:
+        ...
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = ...,
+        event_type: Optional[EventType] = ...,
+        presets: Optional[List[KeywordPresetType]] = ...,
+        notify_channel: Optional[GuildChannel] = ...,
+        timeout_seconds: Optional[int] = ...,
+        enabled: Optional[bool] = ...,
+        exempt_roles: Optional[List[Role]] = ...,
+        exempt_channels: Optional[List[GuildChannel]] = ...,
+        reason: Optional[str] = ...,
+    ) -> AutoModerationRule:
+        ...
+
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = MISSING,
+        event_type: Optional[EventType] = MISSING,
+        keyword_filters: Optional[List[str]] = MISSING,
+        presets: Optional[List[KeywordPresetType]] = MISSING,
+        notify_channel: Optional[GuildChannel] = MISSING,
+        timeout_seconds: Optional[int] = MISSING,
+        enabled: Optional[bool] = MISSING,
+        exempt_roles: Optional[List[Role]] = MISSING,
+        exempt_channels: Optional[List[GuildChannel]] = MISSING,
+        reason: Optional[str] = MISSING,
+    ) -> AutoModerationRule:
+        """
+        |coro|
+        Edit this auto moderation rule.
+
+        You must have the :attr:`~Permissions.manage_guild` to be able to do this.
+
+        Parameters
+        -----------
+        name: Optional[:class:`str`]
```

Comment:
docs should be updated if the above parameter suggestion is applied

Reply:
it would cause confusion through. ill create a note on this maybe

Reply:
nvm